llvm/lib/Transforms/Utils/SimplifyCFG.cpp

```diff
@@ -7882,6 +7882,14 @@ static bool passingValueIsAlwaysUndefined(Value *V, Instruction *I, bool PtrValu
       case Instruction::CallBr:
       case Instruction::Invoke:
         return true;
+      case Instruction::UDiv:
+      case Instruction::URem:
+        // Note: signed div/rem of INT_MIN / -1 is also immediate UB, not
+        // implemented to avoid code complexity as it is unclear how useful such
+        // logic is.
+      case Instruction::SDiv:
+      case Instruction::SRem:
+        return C->isNullValue();
```

Comment:
Don't think you need this check? Division by undef is also UB (if undef is chosen as 0).

Reply:
Don't we need to worry about undef being chosen on non-zero?

Reply:
As undef is chosen per-use, we can always chose zero here, even if another use is non-zero.

Reply:
Done :)